### ISSUE TITLE
Exclude cases with for-loops in ListLiterals from `avoid_single_child` warnings

### ISSUE THREAD

**Issue by naipaka** [altive/altive_lints#42]
The `avoid_single_child` rule should exclude cases where a `for` loop is used within a `ListLiteral`. For instance, the following usage of a `for` loop within the `children` property of a `Column` should not trigger the warning:

```dart
Column(
  children: [
    for (var i = 0; i < 10; i++) Text('Hello $i'),
  ],
),
```

This change ensures that dynamically generated lists using loops are not unnecessarily flagged by the rule.

- for
- if

**Comment by riscait** [comment]
Since collection-if does not increase the number of elements, it seems that only collection-for should be supported.
